diff --git a/pathauto.module b/pathauto.module
index 9bb132d..9838add 100644
--- a/pathauto.module
+++ b/pathauto.module
@@ -438,6 +438,12 @@ function pathauto_entity_language($entity_type, $entity, $check_language_propert
 }
 
 function pathauto_is_alias_reserved($alias, $source, $langcode = LANGUAGE_NONE) {
+  if (module_exists('redirect')) {
+    $rid = redirect_load_by_source($alias, $langcode);
+    if (!empty($rid)) {
+      return TRUE;
+    }
+  }
   foreach (module_implements('pathauto_is_alias_reserved') as $module) {
     $result = module_invoke($module, 'pathauto_is_alias_reserved', $alias, $source, $langcode);
     if (!empty($result)) {
