diff --git a/js/powertagging.field.js b/js/powertagging.field.js
index da18e12..3f3bbeb 100644
--- a/js/powertagging.field.js
+++ b/js/powertagging.field.js
@@ -3,7 +3,7 @@ Drupal.behaviors.powertagging_field = {
   attach: function (context, settings) {
 
     $("div.field-type-powertagging").once(function() {
-      $(this).find("input[type=\"submit\"]").click(function(event) {
+      $(this).find("input[type=\"submit\"],button").click(function(event) {
         event.preventDefault();
         var field_id = $(this).parent().parent().parent().attr("id").substr(5).replace(/-/g, "_");
         var html_id = "#edit-" + field_id.replace(/_/g, "-");
