diff --git a/event_calendar.admin.inc b/event_calendar.admin.inc
index ece8aed..09af891 100755
--- a/event_calendar.admin.inc
+++ b/event_calendar.admin.inc
@@ -38,6 +38,13 @@ function event_calendar_settings($form, $form_state) {
     '#collapsed' => TRUE,
     '#description' => t('Edit the e-mail messages sent to administrator/s. Available variables are: [site:name], [site:url], [node:author], [node:created], [node:title], [node:body], [node:url], [event_calendar:start_date], [event_calendar:end_date], [event_calendar:event_status].'),
   );
+  $form['event_calendar_email_admin']['event_calendar_send_email_admin'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Send to admin'),
+    '#return_value' => 1,
+    '#default_value' => variable_get('event_calendar_send_email_admin', 1),
+    '#description' => t('Check this if you want to send the e-mail to administrators'),
+  );
   $form['event_calendar_email_admin']['event_calendar_email_admin_subject'] = array(
     '#type' => 'textfield',
     '#title' => t('Subject'),
@@ -59,6 +66,13 @@ function event_calendar_settings($form, $form_state) {
     '#collapsed' => TRUE,
     '#description' => t('Edit the e-mail messages sent to selected user roles. Available variables are: [site:name], [site:url], [node:author], [node:created], [node:title], [node:body], [node:url], [event_calendar:start_date], [event_calendar:end_date], [event_calendar:event_status].'),
   );
+  $form['event_calendar_email_user']['event_calendar_send_email_user'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Send to User'),
+    '#return_value' => 1,
+    '#default_value' => variable_get('event_calendar_send_email_user', 1),
+    '#description' => t('Check this if you want to send the e-mail to selected user roles'),
+  );
   $form['event_calendar_email_user']['event_calendar_email_user_subject'] = array(
     '#type' => 'textfield',
     '#title' => t('Subject'),
diff --git a/event_calendar.module b/event_calendar.module
index 1cd24e2..2cdd0c0 100755
--- a/event_calendar.module
+++ b/event_calendar.module
@@ -93,8 +93,9 @@ function event_calendar_node_update($node) {
   $node_type = variable_get('event_calendar_node_type', 'event_calendar');
   if ($node->type == $node_type) {
     $author = user_load($node->uid);
+    $send_to_admin = variable_get('event_calendar_send_email_admin', 1);
     $to = $author->mail;
-    if (!empty($to)) {
+    if (!empty($to) && $send_to_admin == 1) {
       $from = (string) variable_get('site_mail', '');
       $language = language_default();
       $subject = _event_calendar_email_text('users_subject');
@@ -131,21 +132,27 @@ function _event_calendar_email_reciepents($key) {
   $to = array();
   switch ($key) {
     case 'admin':
-      if (in_array($admin_rid, $all_reciepents)) {
-        $to = fetch_emails($admin_rid);
-      }
-      else {
-        $to = fetch_first_emails($admin_rid);
+      $send_to_admin = variable_get('event_calendar_send_email_admin', 1);
+      if ($send_to_admin == 1) {
+          if (in_array($admin_rid, $all_reciepents)) {
+            $to = fetch_emails($admin_rid);
+          }
+          else {
+            $to = fetch_first_emails($admin_rid);
+          }
       }
       break;
 
     case 'users':
-      if (in_array($admin_rid, $all_reciepents)) {
-        unset($all_reciepents[$admin_rid]);
-      }
-      foreach ($all_reciepents as $rid) {
-        $emails = fetch_emails($rid);
-        $to = array_merge($to, $emails);
+      $send_to_users = variable_get('event_calendar_send_email_user', 1);
+      if ($send_to_users == 1) {
+          if (in_array($admin_rid, $all_reciepents)) {
+            unset($all_reciepents[$admin_rid]);
+          }
+          foreach ($all_reciepents as $rid) {
+            $emails = fetch_emails($rid);
+            $to = array_merge($to, $emails);
+          }
       }
       break;
     }
