diff --git a/menu_position.module b/menu_position.module
index 13a11be..8ea4b77 100644
--- a/menu_position.module
+++ b/menu_position.module
@@ -337,6 +337,10 @@ function menu_position_activate_rule($rule, $context, $set_breadcrumb) {
     $preferred_links = &drupal_static('menu_link_get_preferred');
     $preferred_links[$_GET['q']][MENU_PREFERRED_LINK] = menu_link_get_preferred($menu_item['link_path']);
 
+    // Reset static trail and breadcrumb caches.
+    drupal_static_reset('menu_set_active_trail');
+    drupal_static_reset('drupal_set_breadcrumb');
+
     // Remove the menu position router from the end of the trail.
     $active_trail = menu_set_active_trail();
     array_pop($active_trail);
